src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -883,9 +885,11 @@ ValidatorList::applyListsAndBroadcast(
     if (disposition == ListDisposition::accepted)
     {
         bool good = true;
-        for (auto const& [pubKey, listCollection] : publisherLists_)
+
+        // localPublisherList is always available. No need to perform the
+        // below check
```

Comment:
`localPublisherList` contains the information from config file's `SECTION_VALIDATORS`. This is handled in the third (and last portion) of the `ValidatorList::load()` function.

I'm given to understand that the validator information in the local config file does not expire and is always available. Consequently, it is not subject to stringent validity checks.

Note, in the current codebase, such ValidatorLists are mapped to the default constructed Public Key.
I think the comment is correct, perhaps it could be paraphrased to be more informative.

Reply:
> If the local configuration file does not specify a [validators] section, then localPublisherList.list.empty() == true. This maintains the invariant that all the locally configured PublisherList's are perennially available.
>
> I hope this clears things up

Okay.  Now I think I get what's bothering me about this comment.  Bear with me.  You'll find this code around line 1870 of ValidatorList.cpp:
```
        // Remove if expired
        // ValidatorLists specified in the local config file never expire.
        // Hence, the below steps are not relevant for localPublisherList
        if (collection.status == PublisherStatus::available &&
            collection.current.validUntil <= closeTime)
        {
            removePublisherList(lock, pubKey, PublisherStatus::expired);
            ops.setUNLBlocked();
        }
```
What this code shows to me is that there may be validator lists that expire.  I agree with you that a `localPublisherList` cannot expire.  But other validator lists can.

In light of this, your comment says ...
```
        // localPublisherList is always available. No need to perform the
        // below check
```
To me, at least, the closing sentence seems definitive: `No need to perform the below check`.  I think your intention is that there's no need to perform the below check _on the `localPublisherList`_.

I think I would not have been baffled by the comment if it read, for example, like this:
```
        // localPublisherList never expires, so localPublisherList is excluded
        // from the below check.
```
Am I getting closer?  Or am I still misunderstanding something?

Reply:
@ckeshava, thoughts on this comment?  I still think I would have understood your intent better with a comment like this:
```
        // localPublisherList never expires, so localPublisherList is excluded
        // from the below check.
```
But that quibble is not enough reason to delay this pull request further.